send-transaction-service/src/transaction_client.rs

```diff
@@ -230,27 +228,20 @@ where
 ///   scheduler. Most of the complexity of this structure arises from this
 ///   functionality.
 #[derive(Clone)]
-pub struct TpuClientNextClient<T>
-where
-    T: TpuInfoWithSendStatic + Clone,
-{
+pub struct TpuClientNextClient {
     runtime_handle: Handle,
     sender: mpsc::Sender<TransactionBatch>,
     // This handle is needed to implement `NotifyKeyUpdate` trait. It's only
     // method takes &self and thus we need to wrap with Mutex.
     join_and_cancel: Arc<Mutex<(Option<TpuClientJoinHandle>, CancellationToken)>>,
-    leader_updater: SendTransactionServiceLeaderUpdater<T>,
     leader_forward_count: u64,
 }
 
 type TpuClientJoinHandle =
-    TokioJoinHandle<Result<TransactionStatsAndReceiver, ConnectionWorkersSchedulerError>>;
+    TokioJoinHandle<Result<ConnectionWorkersScheduler, ConnectionWorkersSchedulerError>>;
 
-impl<T> TpuClientNextClient<T>
-where
-    T: TpuInfoWithSendStatic + Clone,
-{
-    pub fn new(
+impl TpuClientNextClient {
+    pub fn new<T>(
```

Comment:
I see why the structure is not needed to be generic any more: because Scheduler is stateful and hence generic can be moved just to `new`.